File: bootstrap/src/integrationTest/resources/suite/func/0002.hs
local log = require('log')
local sys = require('sys')
local test = require('test')

local funcs = sys.func.list()
test.assert(funcs.length == 0)

local func_one_id = sys.func.create({
    name = 'func-1'
})

funcs = sys.func.list()
log.info(funcs.length == 1)

local func_one = funcs[1]
test.assert(func_one.id == func_one_id)
test.assert(func_one.name == 'func-1')

local func_two_id = sys.func.create({ })

funcs = sys.func.list()
test.assert(funcs.length == 2)

func_two = funcs[1]
test.assert(func_two.id == func_two_id)
test.assert(func_two.name == '')

func_one = funcs[2]
test.assert(func_one.id == func_one_id)
test.assert(func_one.name == 'func-1')

test.complete()

File: platform/mono/src/integrationTest/resources/api/func/0002.lua
admin = require('sys')

err, funcs = admin.func.list()
assert(err == nil)
assert(#funcs == 0)

err, func_one_req = admin.func.create({ name = 'func-1' })
admin.await_completed(func_one_req)

assert(err == nil)
assert(func_one_req ~= nil)
--
_, funcs = admin.func.list()
assert(#funcs == 1)

assert(func_one_req.id == funcs[1].id)
assert(funcs[1].name == 'func-1')

err, func_two_req = admin.func.create({ name = 'func-2' })
admin.await_completed(func_two_req)

_, funcs = admin.func.list()
assert(#funcs == 2)
